presto-orc/src/main/java/com/facebook/presto/orc/reader/ListStreamReader.java

```diff
@@ -97,45 +100,55 @@ public Block readBlock(Type type)
             }
         }
 
-        // The length vector could be reused, but this simplifies the code below by
-        // taking advantage of null entries being initialized to zero.  The vector
-        // could be reinitialized for each loop, but that is likely just as expensive
-        // as allocating a new array
-        int[] lengthVector = new int[nextBatchSize];
-        boolean[] nullVector = new boolean[nextBatchSize];
-        if (presentStream == null) {
-            if (lengthStream == null) {
-                throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is not present");
-            }
-            lengthStream.nextIntVector(nextBatchSize, lengthVector);
+        // Try to reuse lengthVector, and avoid allocating too large of an array
+        int subBatchSize = Math.min(nextBatchSize, MAX_BATCH_SIZE);
+        if (lengthVector.length < subBatchSize) {
+            lengthVector = new int[subBatchSize];
         }
-        else {
-            int nullValues = presentStream.getUnsetBits(nextBatchSize, nullVector);
-            if (nullValues != nextBatchSize) {
+        Arrays.fill(lengthVector, 0);
+
+        //nullVector and offsetsVector will be passed to the ArrayBlock directly
+        boolean[] nullVector = new boolean[nextBatchSize];
+        int[] offsetsVector = new int[nextBatchSize + 1];
```

Comment:
I have comment on `MapStreamReader` about using the  `offsetsVector` for the length vector since we need to create this array anyway for the block creation.  We can do the same here, and for this code, I think it will be much simpler.